src/test/java/BaseTest.java

```diff
@@ -1,11 +1,357 @@
+    }
+
+    //Play song helper functions
+//    public void chooseAllSongsList()
+//    {
+//        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.cssSelector(".overlay.loading")));
+//        wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("li a.songs"))).click();
+//    }
+
+    public void contextClickFirstSong()
+    {
+        WebElement firstSongElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".all-songs tr.song-item:nth-child(4)")));
+        actions.contextClick(firstSongElement).perform();
+    }
+
+    public void choosePlayOption()
+    {
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("li.playback"))).click();
+    }
+
+    public WebElement hoverPlay()
+    {
+        WebElement play = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[data-testid='play-btn']")));
+        actions.moveToElement(play).perform();
+        return driver.findElement(By.cssSelector("[data-testid='play-btn']"));
+    }
+
+    //Counting number of songs in a playlist
+    public void choosePlaylistByName(String playlistName)
+    {
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[contains(text(),'" + playlistName + "')]"))).click();
+    }
+
+    public int countSongsInPlaylist()
+    {
+        return driver.findElements(By.cssSelector("section#playlistWrapper td.title")).size();
+    }
+
+    public String getPlaylistDetails()
+    {
+        return driver.findElement(By.cssSelector("span.meta.text-secondary span.meta")).getText();
+    }
+    public void displayAllSongs()
+    {
+        List<WebElement> songList = driver.findElements(By.cssSelector("section#playlistWrapper td.title"));
+        System.out.println("Number of Songs Found: " + countSongsInPlaylist());
+        for (WebElement e: songList){
+            System.out.println(e.getText());
+        }
+    }
+
+    //Helper functions to rename playlist
+
+    String enterNewPlaylistName = "My New Playlist";
+    public void doubleClickPlaylist()
+    {
+        WebElement playlistElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".playlist:nth-child(3)")));
+        actions.doubleClick(playlistElement).perform();
+    }
+
+    public void enterNewPlaylistName()
+    {
+        WebElement playListInputField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[name='name']")));
+        playListInputField.sendKeys(Keys.chord(Keys.CONTROL, "A", Keys.BACK_SPACE));
+        playListInputField.sendKeys(enterNewPlaylistName);
+        playListInputField.sendKeys(Keys.ENTER);
+    }
+
+    public boolean doesPlaylistNameExist()
+    {
+        WebElement playlistElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[contains(text(),'" + enterNewPlaylistName + "')]")));
+        return playlistElement.isDisplayed();
+    }
```

Comment:
Page Object Model (POM): recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

Move these methods to the corresponding class according to the POM.